packages/react/src/hooks/utils/useChainId.ts

```diff
@@ -1,6 +1,10 @@
 import { useProvider } from '../providers'
 
-export function useChainId() {
-  const provider = useProvider()
-  return provider?.network?.chainId
+export type UseChainIdArgs = {
+  chainId?: number
+}
+
+export function useChainId({ chainId }: UseChainIdArgs = {}) {
```

Comment:
Can the returned chain id be different to the passed `chainId`?

Reply:
It can, but only if the provider does not support the chain. [Added a test.](https://github.com/tmm/wagmi/pull/330/files)